backend/open_webui/models/functions.py

```
@@ -146,52 +275,103 @@ def get_functions_by_type(
                 .all()
             ]
 
-    def get_function_valves_by_id(self, id: str) -> Optional[dict]:
-        with get_db() as db:
+    def get_function_valves_by_id(
+        self, id: str, db: Optional[Session] = None
+    ) -> Optional[dict]:
+        with get_db_context(db) as db:
             try:
                 function = db.get(Function, id)
                 return function.valves if function.valves else {}
```

Ruff (0.15.4)

[warning] Consider moving this statement to an `else` block

(TRY300)